### PR TITLE
Update to allow PHPUnit 4 or 5

### PR DESCRIPTION
**Pull request by jimlind**
Allow PHPUnit 4 or 5.
Update ReadMe.

### CODE REVIEW

**Comment by malkusch** [comment]
Could you please squash PR into one commit?

File: composer.json [hunk 1]
```diff
@@ -17,8 +17,8 @@
         "psr-4": {"phpmock\\phpunit\\": "classes/"}
     },
     "require": {
-        "php": ">=5.5",
-        "phpunit/phpunit": "^4.0.0",
+        "php": ">=5.6",
```
Comment (samsonasik) [inline composer.json]: it should can still >=5.5 imo

File: composer.json [hunk 2]
```diff
@@ -17,8 +17,8 @@
...
+        "phpunit/phpunit": "^4.0.0 || ^5.0.0",
```
Comment (samsonasik) [inline composer.json]: it should ~4.0|~5.0

File: .travis.yml [hunk 3]
```diff
@@ -21,7 +21,6 @@ env:
 php:
   - 7.0
   - 5.6
-  - 5.5
```
Comment (samsonasik) [inline .travis.yml]: still support version 4. Means 5.5 should still supported imo, althought phpunit4 support php 5.4,
There is no reason to drop 5.5

**Comment by malkusch** [comment]
Regarding PHP-5.5 I added already a follow up commit 38973105d6a75a6bd0474bc04584b6543fe457fe to support PHP-5.5 again. I just wait for a squash of this PR and would add PHP-5.5 back.

File: composer.json (hunk 2 above)
Comment (malkusch) [inline composer.json]: `"^4.0.0 || ^5.0.0"` is fine.

File: composer.json (hunk 1 above)
Comment (malkusch) [inline composer.json]: Regarding PHP-5.5 I added already a follow up commit 38973105d6a75a6bd0474bc04584b6543fe457fe to support PHP-5.5 again. I just wait for a squash of this PR and would add PHP-5.5 back.

File: .travis.yml (hunk 3 above)
Comment (malkusch) [inline .travis.yml]: Don't worry, it won't be dropped before 10 Jul 2016 ;)

Comment (jlind) [inline .travis.yml]: I did get a little overzealous upgrading. 5.4 is still supported

**Comment by jimlind** [comment]
Squashed per @malkusch.
Allow PHP 5.5 per @samsonasik.

**Pull request merged**